webapp/channels/src/components/form_error.tsx

```diff
@@ -11,71 +11,70 @@ type Props = {
     iconClassName?: string;
     margin?: boolean;
     errors?: React.ReactNode[];
-}
+};
 
-export default class FormError extends React.PureComponent<Props> {
-    public static defaultProps = {
-        error: null,
-        errors: [],
-    };
-    public render() {
-        const {error = null, errors = [], iconClassName, margin, textClassName, type} = this.props;
-
-        if (!error && errors.length === 0) {
-            return null;
-        }
+const FormError: React.FC<Props> = ({
+    error = null,
+    errors = [],
```

Comment:
By using a list here, when a parent re-renders this component, `errors` will always have a new value.

This will not create new re-renders of `FormError`, since the parent will always send the same prop (undefined) but `FormError` re-rendering for other reasons (a margin change?) will create a new `errors` list.

This doesn't create so much issue right now, since we don't expect this to be re-rendered that often. Also, `errors` is processed inside the body of the component and never assigned to dependencies or children props.

But in order to be future proof, and make sure any future change in the component does not suffer from that value being re-generated unconsciously, we can define the empty list as a constant outside of the component. Something like this:

```
const EMPTY_ERROR_LIST: React.ReactNode[]  = [];

const FormError: React.FC<Props> = ({
    errors = EMTPY_ERROR_LIST;
    ...
```

Reply:
Done all suggested changes! Thank you for great explaination